### DISCUSSION TITLE
In memory policy

### DISCUSSION THREAD

**Discussion by d1ss0nanz** [casbin/casbin#1159] [Q&A] [answered]
When an application always generates the policy at startup and the policy doesn't change after that at runtime, do we need a persistance layer?

I'd like to do something like:
`casbin.NewEnforcer(authorizationModelConfig, "")`
but that doesnt' work.

I've found the following in the code:
`
	if err = e.adapter.LoadPolicy(newModel); err != nil && err.Error() != "invalid file path, file path cannot be empty" {
		return err
	}
`
Execution still fails with "invalid file path, file path cannot be empty".

A workaround is providing an empty, but existing file via the file adapter.

Am I missing something?

**Comment by hsluoyz** [comment, marked as answer]
@d1ss0nanz create github issue instead
